rasa/nlu/components.py

```diff
@@ -68,89 +67,64 @@ def validate_tokenizers(pipeline: List["Component"]) -> None:
         )
 
 
-def validate_required_components(pipeline: List["Component"]) -> None:
-    """Validates that all required components are present in the pipeline."""
-
-    unique_component_names = set()
-    for component in pipeline:
-        unique_component_names.add(component.name)
-        if not set(component.required_components).issubset(unique_component_names):
-            raise InvalidConfigError(
-                f"'{component.name}' requires {component.required_components}. "
-                f"Add required components to the pipeline."
-            )
-
-
-def validate_arguments(
-    pipeline: List["Component"],
-    context: Dict[Text, Any],
-    allow_empty_pipeline: bool = False,
-) -> None:
-    """Validates that all arguments are present to train the pipeline."""
-
-    # Ensure the pipeline is not empty
-    if not allow_empty_pipeline and len(pipeline) == 0:
```

Comment:
Should we keep the check for an empty pipeline?

Reply:
I don't know, why did we add it?

Reply:
yes, thanks for spotting it